private/bufpkg/bufconnect/interceptors_test.go

```diff
@@ -21,18 +21,70 @@ import (
 	"testing"
 
 	"github.com/bufbuild/buf/private/pkg/app"
+	"github.com/bufbuild/buf/private/pkg/netrc"
 	"github.com/bufbuild/connect-go"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestAuthorizationInterceptorProviderTokenErr(t *testing.T) {
-	_, err := NewAuthorizationInterceptorProvider(app.NewEnvContainer(map[string]string{
-		tokenEnvKey: "test1234",
-	}))("fake")(func(context.Context, connect.AnyRequest) (connect.AnyResponse, error) {
+	// test setting auth token with provided token
+	_, err := NewAuthorizationInterceptorProvider(AuthorizeWithProvidedToken("test1234"))("fake")(func(context.Context, connect.AnyRequest) (connect.AnyResponse, error) {
 		return nil, errors.New("underlying cause")
 	})(context.Background(), connect.NewRequest(&bytes.Buffer{}))
+	assert.Error(t, err)
 
-	authErr, ok := AsAuthError(err)
-	assert.True(t, ok)
-	assert.Equal(t, tokenEnvKey, authErr.tokenEnvKey)
+	// test on using remote token
+	container := app.NewEnvContainer(map[string]string{
+		tokenEnvKey: "username:token@remote,buftoken",
+	})
+	_, err = NewAuthorizationInterceptorProvider(AuthorizeWithAddress(container))("remote")(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		if req.Header().Get(AuthenticationHeader) != AuthenticationTokenPrefix+"token" {
+			return nil, errors.New("error auth token found")
+		}
+		return nil, nil
+	})(context.Background(), connect.NewRequest(&bytes.Buffer{}))
+	assert.NoError(t, err)
+
+	// test on using default token
+	container = app.NewEnvContainer(map[string]string{
+		tokenEnvKey: "username:token@remote,buftoken",
+	})
+	_, err = NewAuthorizationInterceptorProvider(AuthorizeWithAddress(container))("fake")(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		if req.Header().Get(AuthenticationHeader) != AuthenticationTokenPrefix+"buftoken" {
+			return nil, errors.New("error auth token found")
+		}
+		return nil, nil
+	})(context.Background(), connect.NewRequest(&bytes.Buffer{}))
+	assert.NoError(t, err)
+
+	// test on zero value
+	getMachineForName = newGetMachineForName
+	container = app.NewEnvContainer(map[string]string{})
+	_, err = NewAuthorizationInterceptorProvider(AuthorizeWithAddress(container))("fake")(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		if req.Header().Get(AuthenticationHeader) != AuthenticationTokenPrefix+"password" {
+			return nil, errors.New("error auth token found")
+		}
+		return nil, nil
+	})(context.Background(), connect.NewRequest(&bytes.Buffer{}))
+	assert.NoError(t, err)
+	getMachineForName = netrc.GetMachineForName
+}
```

Comment:
@tonyli233 , could we split this up into multiple test functions. It looks like it is executing multiple tests. The previous version of the test was very compact and easy to grok, not it is quite large.

Reply:
the tests are changed according to the new interceptor constructor